harness/determined/deploy/aws/cli.py

```diff
@@ -172,10 +172,22 @@ def deploy_aws(command: str, args: argparse.Namespace) -> None:
                 f"--db-size cannot be specified for deployment types other than "
                 f"{constants.deployment_types.SIMPLE_RDS} got {args.deployment_type}"
             )
+        if args.db_snapshot != constants.defaults.DB_SNAPSHOT:
+            raise ValueError(
+                f"--db-snapshot cannot be specified for deployment types other than "
+                f"{constants.deployment_types.SIMPLE_RDS} got {args.deployment_type}"
+            )
     else:
         if args.db_size is not None and args.db_size < 20:
             raise ValueError("--db-size must be greater than or equal to 20 GB")
 
+        if args.db_snapshot is not None and args.db_password != constants.defaults.DB_PASSWORD:
+            # this precedence is implemented in the deployment template
+            print(
```

Comment:
Mutually exclusive sounds like it should be an error. Perhaps we can just mention the precedence?